app/middleware/validateUrl.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const validateUrl = (req) => {
+    const redirect = req.session.returnUrl || '/offline';
+    let givenUrl ;
+    try {
+        givenUrl = new URL (redirect);
+    } catch (error) {
+        req.log.error(error);
+        givenUrl = '/offline';
```

Comment:
Please replace this string with URL instance, so that return type is the same and not "URL" if a valid and String if invalid.

Reply:
If we change this '/offline' to URL object it will give error until we have full correct url

Reply:
Can't change 'offline' string to URL object as it will give Invalid Url error until we get the complete path . Don't think that's needed here